port/unix/omrsysinfo.c

```diff
@@ -6719,41 +6821,62 @@ omrsysinfo_cgroup_subsystem_iterator_next(struct OMRPortLibrary *portLibrary, st
 			state->count += 1;
 			goto _end;
 		}
-
-		state->fileMetricCounter = 0;
 	}
 
 	if (state->fileMetricCounter < subsystemMetricMapElement->metricElementsCount) {
-		BOOLEAN isMetricFound = FALSE;
 		char *current = state->fileContent;
-		char *end = state->fileContent + strlen(state->fileContent) - 1;
-		int32_t metricKeyLen = 0;
 		currentElement += state->fileMetricCounter;
-		metricKeyLen = strlen(currentElement->metricKeyInFile);
-		while (current < end) {
-			if (0 == strncmp(current, currentElement->metricKeyInFile, metricKeyLen)) {
-				char *newLine = current;
-				current += metricKeyLen;
-				/* advance past any whitespace */
-				while ((current != end) &&  (' ' == *current)) {
+
+		if ((OMR_ARE_ANY_BITS_SET(PPG_sysinfoControlFlags, OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE))
+			&& (CGROUP_METRIC_NO_KEY == (uintptr_t)currentElement->metricKeyInFile)
```

Comment:
In continuation of the previous comment, v1/v2 support is checked in the opposite order; if both are available will this cause a problem?

Reply:
See above reply, also `OMRPORT_SYSINFO_CGROUP_V1_AVAILABLE` and `OMRPORT_SYSINFO_CGROUP_V2_AVAILABLE` are mutually exclusive